internal/static/metrics/wal/wal.go

```diff
@@ -91,6 +90,11 @@ func newStorageMetrics(r prometheus.Registerer) *storageMetrics {
 		Help: "Total number of exemplars appended to the WAL",
 	})
 
+	m.totalAppendedMetadata = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_remote_write_wal_metadata_appended_total",
+		Help: "Total number of metadata entries appended to the WAL",
+	})
```

Comment:
```suggestion
	m.totalAppendedMetadata = prometheus.NewCounter(prometheus.CounterOpts{
		Name: "prometheus_remote_write_wal_metadata_updates_total",
		Help: "Total number of metadata updates sent through the WAL",
	})
```

Since we aren't necessarily appending Metadata we only do it if it was updated.